src/common/reduction/analysis/test/EventAnalyzerTest.cpp

```diff
@@ -85,45 +85,17 @@ TEST_F(EventAnalyzerTest, AnalyzeAverage) {
   EXPECT_EQ(result.time, 3);
   
   result = EventAnalyzer("utpc").analyze(cluster);
-  EXPECT_EQ(result.center, 6);
+  EXPECT_EQ(result.center, 5);
   EXPECT_EQ(result.time, 5);
 
   result = EventAnalyzer("utpc_weighted").analyze(cluster);
-  EXPECT_EQ(result.center, 5.5);
-  EXPECT_EQ(result.time, 5);
-  
-
+  EXPECT_GT(result.center, 4.46);
+  EXPECT_LT(result.center, 4.47);
+  EXPECT_EQ(result.time, 4);
 }
 
-TEST_F(EventAnalyzerTest, AnalyzeUncert) {
-  hit.weight = 1;
-
-  hit.time = hit.coordinate = 0;
```

Comment:
I don't do this nearly enough myself, but consider adding some 
comments also to unit tests.

Reply:
Of course not in deleted code, but you know what I mean :-)